praw/models/listing/generator.py

```diff
@@ -39,11 +49,11 @@ def __init__(self, reddit, url, limit=100, params=None):
         self.url = url
         self.yielded = 0
 
-    def __iter__(self):
+    def __iter__(self) -> _ListingGenerator:
```

Comment:
While `__iter__` technically returns itself it is a magic method, so I'm curious how said magic methods are typed in the standard library, or well used third-party libraries that use type hints.

Reply:
According to the PEP, it should be an iterator return

Reply:
Perfect.